### PR TITLE
Added joining tabular data example

### PR DESCRIPTION
**Pull request by danfowler**

### CODE REVIEW

File: guides/joining-data-in-python.md [hunk 1]
```diff
@@ -141,3 +141,101 @@ a country, it's property list will show up featuring "ADMIN",
 "ISO_A3", and the newly added "GDP (2014)" property.
 
 <script src="https://embed.github.com/view/geojson/frictionlessdata/example-data-packages/master/_other_example_files/world_gdp_2014.geojson"></script>
+
+## Joining Two Tabular Datasets
+
+Here's an example of joining two tabular datasets.  In this case, we
+can use a similar GDP dataset (this time with all years) and combine
+it with the Consumer Price Index, a measure of inflation, to yield
+[Real GDP](https://en.wikipedia.org/wiki/Real_gross_domestic_product)
```
Comment (rufuspollock) [inline guides/joining-data-in-python.md:150]: I would like to recommend much reducing this kind of external linkages. It is a bit confusing to readers (should i follow this link) and is not necessary to follow the article.

File: guides/joining-data-in-python.md [hunk 2]
```diff
@@ -141,3 +141,101 @@ a country, it's property list will show up featuring "ADMIN",
...
+of 'number'.
+
+{% highlight python %}
+field_info = [f for f in cpi_dp.resources[0].metadata['schema']['fields']]
+field_info_ext = [f for f in gdp_dp.resources[0].metadata['schema']['fields']]
+field_info.extend(field_info_ext)
+field_info.append({'name': 'Real GDP', 'type': 'number'})
+{% endhighlight %}
+
+Now that we have this information, we can generate a `fieldnames`
```
Comment (rufuspollock) [inline guides/joining-data-in-python.md:188]: This is the sort of thing we would want automated. Generating the _new_ Data Package should be thing we do and then writing out is basically automatic.

**Comment by rufuspollock** [comment]
@danfowler generally looks good. Can you allow a moment for review before we merge - a nice practice to get into 😉

**Comment by pwalsh** [comment]
@danfowler would be good if you ask for reviews for PRs, just like with all other tech work. It will also allow internal team to get visibility into what you are doing, and to give early feedback :)

**Comment by rufuspollock** [comment]
@danfowler main points:
- I think we want this as a separate tutorial rather than all in one
- I think this is useful start in showing exactly _what_ we would like automated. It seems clear reading this that I want something like:

```
dp1 = DataPackage('gdp-url')
dp2 = DataPackage('price-index-url')

dp3 = DataPackageTools.join(dp1, dp2, ['Country', 'Year'])

dp3.save('path')
```

**Comment by danfowler** [comment]
@rgrp sure, I'll split out into a separate tutorial.

@rgrp @pwalsh yes, will ask for reviews in the future 👍

**Comment by danfowler** [comment]
@rgrp `dp3` should reference the join column(s) on both `dp1` and `dp2`:

```
dp3 = DataPackageTools.join(dp1, dp2, {['Country', 'Country'],['Year','Year']})
```

File: guides/joining-data-in-python.md (hunk 1 above)
Comment (danfowler) [inline guides/joining-data-in-python.md:150]: Noted.  Reducing.

**Comment by rufuspollock** [comment]
@danfowler great to see this merged. Can you make a summary of learnings (including my comments) on the main issue related to this in ideas issue tracker.

**Comment by danfowler** [comment]
@rgrp some commentary already in the thread as replies to the original idea: https://github.com/frictionlessdata/ideas/issues/257

**Pull request merged**